napari/layers/_base_layer/model.py

```diff
@@ -281,17 +280,6 @@ def scale_factor(self):
 
         return scale_factor[0]
 
-    def _after_set_viewer(self, prev):
-        """Triggered after a new viewer is set.
-
-        Parameters
-        ----------
-        prev : Viewer
-            Previous viewer.
-        """
-        if self.viewer is not None:
-            self.refresh()
```

Comment:
I don't think that this will break anything now but it might be good to keep track of in case anyone wants to juggle layers across multiple viewers!

Reply:
In a soon to come PR the `viewer` property will be removed entirely from the `Layer` so as to prevent circular referencing. I'm not quite sure at that point how moving layers from one viewer to another would work. `viewer.add_layer(layer)` seems like find syntax, but we'd have to make sure that worked. I'm not sure it that is supported now